client-spark/spark3/src/main/java/org/apache/spark/shuffle/writer/RssShuffleWriter.java

```diff
@@ -473,4 +492,61 @@ Map<Integer, Set<Long>> getPartitionToBlockIds() {
   public WriteBufferManager getBufferManager() {
     return bufferManager;
   }
+
+  private static ShuffleManagerClient createShuffleManagerClient(String host, int port)
+      throws IOException {
+    ClientType grpc = ClientType.GRPC;
+    // Host can be inferred from `spark.driver.bindAddress`, which would be set when SparkContext is
+    // constructed.
+    return ShuffleManagerClientFactory.getInstance().createShuffleManagerClient(grpc, host, port);
```

Comment:
Can we reduce the number of ShuffleManagerClient creations?

Reply:
> Can we reduce the number of ShuffleManagerClient creations?

I wrote it in the way that Read failed, and I will fix it together in the next PR.

Reply:
Okay.